packages/modules/src/BatchedSessionRouterModule.ts

```diff
@@ -0,0 +1,290 @@
+   * @param moduleConfig The configuration for the module
+   * @returns An instance of SessionKeyManagerModule
+   */
+  private constructor(moduleConfig: BatchedSessionKeyManagerModuleConfig) {
+    super(moduleConfig)
+  }
+
+  /**
+   * Asynchronously creates and initializes an instance of SessionKeyManagerModule
+   * @param moduleConfig The configuration for the module
+   * @returns A Promise that resolves to an instance of SessionKeyManagerModule
+   */
+  public static async create(
+    moduleConfig: BatchedSessionKeyManagerModuleConfig
+  ): Promise<BatchedSessionRouterModule> {
+    const instance = new BatchedSessionRouterModule(moduleConfig)
+
+    if (moduleConfig.moduleAddress) {
+      instance.moduleAddress = moduleConfig.moduleAddress
+    } else if (moduleConfig.version) {
+      const moduleAddr = BATCHED_SESSION_ROUTER_MODULE_ADDRESSES_BY_VERSION[moduleConfig.version]
+      if (!moduleAddr) {
+        // throw new Error(`Invalid version ${moduleConfig.version}`)
+        Logger.error(`Invalid version ${moduleConfig.version}`)
+        instance.moduleAddress = DEFAULT_BATCHED_SESSION_ROUTER_MODULE
+      }
+      instance.moduleAddress = moduleAddr
+      instance.version = moduleConfig.version as ModuleVersion
+    }
+
+    instance.sessionManagerModuleAddress =
+      moduleConfig.sessionManagerModuleAddress ?? DEFAULT_SESSION_KEY_MANAGER_MODULE
+    instance.nodeClient = new NodeClient({
+      txServiceUrl: moduleConfig.nodeClientUrl ?? NODE_CLIENT_URL
+    })
+
+    if (!moduleConfig.storageType || moduleConfig.storageType === StorageType.LOCAL_STORAGE) {
+      instance.sessionStorageClient = new SessionLocalStorage(moduleConfig.smartAccountAddress)
+    } else {
+      throw new Error('Invalid storage type')
+    }
+
+    const existingSessionData = await instance.sessionStorageClient.getAllSessionData()
+    const existingSessionDataLeafs = existingSessionData.map((sessionData) => {
+      const leafDataHex = hexConcat([
+        hexZeroPad(ethers.utils.hexlify(sessionData.validUntil), 6),
+        hexZeroPad(ethers.utils.hexlify(sessionData.validAfter), 6),
+        hexZeroPad(sessionData.sessionValidationModule, 20),
+        sessionData.sessionKeyData
+      ])
+      return ethers.utils.keccak256(leafDataHex)
+    })
+
+    instance.merkleTree = new MerkleTree(existingSessionDataLeafs, keccak256, {
+      sortPairs: true,
+      hashLeaves: false
+    })
+
+    return instance
+  }
+
+  /**
+   * Method to create session data for any module. The session data is used to create a leaf in the merkle tree
+   * @param leavesData The data of one or more leaves to be used to create session data
+   * @returns The session data
+   */
+  createSessionData = async (leavesData: CreateSessionDataParams[]): Promise<string> => {
+    const sessionKeyManagerModuleABI = 'function setMerkleRoot(bytes32 _merkleRoot)'
+    const sessionKeyManagerModuleInterface = new ethers.utils.Interface([
```

Comment:
Let this call sessionKeyManager module's createSessionData method.

Reply:
ok, we can create an instance of it within this class as suggested in other comment.

Reply:
done